media/src/Settings.svelte

```diff
@@ -115,72 +122,149 @@
-               on:change={rescale}
-               /><span class="output">{scala}</span></span>
+    <label class="form-label" for="scale">Scale</label>
+    <span class="form-range">
+        <input
+            type="range"
+            name="scale"
+            id="scale"
+            min="-2"
+            max="3"
+            step=".02"
+            bind:value={scale}
+            on:change={rescale}
+        >
+        <span class="output box-23 text-right">{scala}</span>
+    </span>
 
-    <span class="box-1 text-right">grid</span>
-    <label class="switch box-2">
+    <div class="form-check form-switch">
+        <label class="form-check-label box-1" for="gridVisible">Grid</label>
         <input
+            class="form-check-input box-1"
             type="checkbox"
             name="gridVisible"
             id="gridVisible"
+            role="switch"
+            aria-checked="true"
             bind:checked={gridMeshes.visible}
             on:change={render}
-            />
-        <span class="slider round" />
-    </label>
+        />
+    </div>
 
-    <span class="box-1 text-right">ortho/proj</span>
-    <label class="switch box-2">
+    <div class="form-check form-switch">
+        <label class="form-check-label box-2" for="orthoCamera">Linear Perspective</label>
         <input
+            class="form-check-input box-2"
             type="checkbox"
+            role="switch"
+            aria-checked="false"
             name="orthoCamera"
             id="orthoCamera"
             bind:checked={orthoCamera}
             on:change={render}
-            />
-        <span class="slider round" />
-    </label>
+        />
+    </div>
+</div>
+
+<div
+    class="settings-box"
+    class:grid={showUpload}
+    hidden={!showUpload}
+    id="upload-box"
+>
+    <button
+        type="button"
+        class="btn btn-sm btn-light box-3"
+        style="margin-left: auto;"
+        on:click={() => {
+            showUpload = false;
+        }}
+    >
+        <i class="bi bi-x-lg"></i>
+    </button>
+    <label for="upload">Upload a scene</label>
+    <input id="upload" type="file" accept="application/json"
+        on:change={(e) => {
+            const reader = new FileReader();
+            reader.onload = (evt) => {
+                try {
+                    let upload = JSON.parse(evt.target.result);
+                    if (upload.length < 33) {
+                        for (let i = 0; i < upload.length; i++) {
+                            objects = makeObject(
+                                null, 
+                                upload[i].kind,
+                                upload[i].params,
+                                objects,
+                                socket
+                            );
+                        }
+                    } else {
+                        alert('Object limit of 32 per upload.');
+                    }
+                } catch (SyntaxError) {
```

Comment:
The "try" part of a try/catch block should only contain the statement which you expect could throw an exception. Having too much code in there can lead to unexpected behavior. I would refactor to something like:

```
                let importedScene = null;
                try {
                    importedScene = JSON.parse(evt.target.result);
                } catch (SyntaxError) {
                    // handle error
                }

                if (importedScene.length < 33) {
                    for (let i = 0; i < importedScene.length; i++) {
                        objects = makeObject(
                            null, 
                            importedScene[i].kind,
                            importedScene[i].params,
                            objects,
                            socket
                        );
                    }
                } else {
                    alert('Object limit of 32 per scene import.');
                }
```

Reply:
My thinking was for the `try` statement to skip the array length check if the upload failed. Changes made, but I'm going to initialize the upload as an empty list since that's what is expected. It would throw an error if you try to check the length of null.